pkg/vault/contracts/Vault.sol

```diff
@@ -376,6 +376,12 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
 
         // (1) and (2): get raw amounts and check limits.
         if (params.kind == SwapKind.EXACT_IN) {
+            // Set locals.swapFeeAmountScaled18 based on the amountCalculated.
+            if (state.swapFeePercentage > 0) {
```

Comment:
I can't comment outside the diff, but L370-375 should be eliminated as we're replacing that with this specific code for each swap kind.

Reply:
already addressed while I was reviewing :)